### ISSUE TITLE
As a developer, the "va" function has been replaced with something more memory safe

### ISSUE THREAD

**Issue by solarisstar** [solarisstar/rpgxEF#56]
Here's the function

``` C
char    * QDECL va(char *format, ...) {
    va_list     argptr;
    static char     string[2][32000];   /* in case va is called by nested functions */
    static int      index = 0;
    char    *buf;

    buf = string[index & 1];
    index++;

    va_start(argptr, format);
    vsprintf(buf, format, argptr);
    va_end(argptr);

    return buf;
}
```

obviously this is not memory safe as it's returning an address on the stack. If it's called multiple times within the same scope (or in a subscope) the current implementation could clobber existing char buffers created with it. The best approach is probably to just pass in a buffer, or straight up replace it with sprintf.

**Comment by solarisstar** [comment]
Also it's terribly named.

**Comment by MJLHThomassen** [comment]
I agree about its naming. I also personally don't see the use of this function and having the string[2][32000] there hardcoded is also begging for problems.

I'm in favour of replacing it with sprintf. They probably didn't do that because of the VM's, but since we're not using them anyway we can use sprintf safely enough.

**Comment by solarisstar** [comment]
It is using vsprintf internally, and Com_Sprintf should work in all these circumstances,

**Comment by solarisstar** [comment]
This function needs 64kb of memory just to exist. That's pretty awful.

**Comment by MJLHThomassen** [comment]
Yeah, if we can replace it with somethign better then lets.

**Comment by solarisstar** [comment]
Not quite sure what the best approach is to replace this in some places. It's not immediately clear in many places what the max size of the buffer should be and I'm hesitant to start guessing. Obviously I can't litter the code with 32kb buffers everywhere or we're going to eat up a ton of stack space.
